.gitignore

```diff
@@ -84,7 +119,80 @@ Release
 DLL Debug
 DLL Release
 
-# Eclipse
+*.trs
+firmware.bin
+wolfmqtt/options.h
+IDE/ARDUINO/wolfMQTT
+IDE/Microchip-Harmony/wolfmqtt_client/firmware/mqtt_client.X/dist/default/
+build/
+
+# eclipse
 .cproject
 .project
 .settings
+
+
+# auto-created CMake backups
+**/CMakeLists.txt.old
+
+# VisualGDB
+**/.visualgdb
+**/*.vgdbproj.*.user
+
+
+# Espressif sdk config default should be saved in sdkconfig.defaults
```

Comment:
The following items are pathed from root, not local to the git repo, why are they here? Git shouldn't see these.